### ISSUE TITLE
extract_diff: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!

### ISSUE THREAD

**Issue by DonStroganotti** [KohakuBlueleaf/LyCORIS#146]
https://github.com/KohakuBlueleaf/LyCORIS/blob/e3aae0c94a4be1c2fdf7aefaa94e76e48b033beb/lycoris/utils/__init__.py#L232C13-L243C22

Manually installed the dev branch for Kohya as the version in requirements.txt doesn't work. 

I don't really know what I am doing but moving the detach call chain outside of the inner parenthesis makes the code run successfully. 

```            
diff = (
      root_weight
      - torch.einsum(
          "i j k l, j r, p i -> p r k l",
          extract_c,
          extract_a.flatten(1, -1),
          extract_b.flatten(1, -1),
      )           
  ).detach().cpu().contiguous()
```

**Comment by KohakuBlueleaf** [comment]
This will be fixed in 2.1.0

**Comment by KohakuBlueleaf** [comment]
Fixed in 2.1.0
